### ISSUE TITLE
data-formatter does not run if it is referencing a function inside of an object

### ISSUE THREAD

**Issue by CowDotDev** [fooplugins/FooTable#502]
I was trying to use `data-formatter` on some columns, and the function I wanted to use was inside of an object (i.e. `App.formatter`). When rendering the table, it appears as though the filter never ran, and if I put a `console.log()` in the function it never fires.

However, if I reference a function that is not in an object it will run just fine and the `console.log()` will fire as many times as there are rows of data.

The object of functions is being declared in the head tag, and the footable initiator is fired after the body loads. So, the object of functions exists before the formatter is ran. Also, there are no errors trying to use the object function.

**Example of it working:**

```
thead
    tr
        th(data-sort-initial="true", data-class="expand", data-type="date", data-format-string="MMMM Do YYYY").footable-sortable Date
        th(data-breakpoints="xs",data-type="text").footable-sortable Status
        th(data-breakpoints="xs",data-type="text").footable-sortable Venue
        th(data-breakpoints="xs sm",data-type="number",data-formatter="formatCurrency",data-decimal-separator=".",data-thousand-separator=",").footable-sortable Subtotal
        th(data-breakpoints="xs sm",data-type="number",data-decimal-separator=".",data-thousand-separator=",").footable-sortable Tax
        th(data-breakpoints="xs sm",data-type="number",data-decimal-separator=".",data-thousand-separator=",").footable-sortable Gratuity
        th(data-type="number",data-decimal-separator=".",data-thousand-separator=",") Total
```

**Example of it NOT working:**

```
thead
    tr
        th(data-sort-initial="true", data-class="expand", data-type="date", data-format-string="MMMM Do YYYY").footable-sortable Date
        th(data-breakpoints="xs",data-type="text").footable-sortable Status
        th(data-breakpoints="xs",data-type="text").footable-sortable Venue
        th(data-breakpoints="xs sm",data-type="number",data-formatter="App.formatCurrency",data-decimal-separator=".",data-thousand-separator=",").footable-sortable Subtotal
        th(data-breakpoints="xs sm",data-type="number",data-decimal-separator=".",data-thousand-separator=",").footable-sortable Tax
        th(data-breakpoints="xs sm",data-type="number",data-decimal-separator=".",data-thousand-separator=",").footable-sortable Gratuity
        th(data-type="number",data-decimal-separator=".",data-thousand-separator=",") Total
```

**Comment by steveush** [comment]
Hi,

At present the function used must be in the global scope which is understandably not the best. Under the hood to avoid using `eval` to parse a dot notated string name such as `App.formatCurrency` I made a very simple [check on the window](https://github.com/fooplugins/FooTable/blob/V3/src/js/utils/FooTable.js#L26) for the function name which is why it is limited to the window. Try out the below version of the function which will handle dot notation method names by simply including it in the page after you have included the main footable files but before you initialize the plugin.

``` javascript
(function(F){

    F.getFnPointer = function(functionName){
        if (F.is.emptyString(functionName)) return null;
        var pointer = window,
            parts = functionName.split('.');
        F.arr.each(parts, function(part){
            if (pointer[part]) pointer = pointer[part];
        });
        return F.is.fn(pointer) ? pointer : null;
    };

})(FooTable);
```

Thanks

**Comment by steveush** [comment]
This change will be rolled into the next version 3.0.12.

**Comment by steveush** [comment]
Due to some backwards compatibility issues with filtering the new version has been released as 3.1.0 and this change has been included.

**Comment by CowDotDev** [comment]
Awesome! Thank you for the help.
